.github/workflows/main.yml

```diff
@@ -0,0 +1,24 @@
+name: CI on Pull Request
+
+on:
+  pull_request:
+    branches:
+      - '*'
+  push:
+    branches:
+      - '*'
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
```

Comment:
I'd suggest adding one step to set up Go to use the same version as in the go.mod file, for example:
```
  - uses: actions/setup-go@v5
    with:
      go-version-file: 'path/to/go.mod'
```
see https://github.com/actions/setup-go